### ISSUE TITLE
feat: work as a drop-in ls replacement

### ISSUE THREAD

**Issue by WhyNotHugo** [eza-community/eza#736]
Currently, `exa` doesn't support the `-t` flag in the same way as `ls`, so it won't work as a drop in replacement. That is, `ls -ltra` works, but `exa -ltra` won't work.

An issue existed at the exa repository for this: https://github.com/ogham/exa/issues/519

I also wrote a patch fixing this: https://github.com/ogham/exa/commit/fa76a335fd40d1f81b656b3d602309e7d39bed2f

I tried cherry-picking this, but some conflicts seem non-trivial. I'd appreciate hearing some feedback on the patch first, and if it looks good, I'll like to incorporate it into eza. Notably:

- When `-t` is specified with arguments, it behaves much like `exa -t X`.
- When `-t` is specified without arguments, it behaves like `ls`.

**Comment by MartinFillon** [comment]
this can either be picked up by someone as I myself will wait/implement this in #638 but not in main branch

**Comment by moonfruit** [comment]
> * When `-t` is specified with arguments, it behaves much like `exa -t X`.
> 
> * When `-t` is specified without arguments, it behaves like `ls`.

I don't think this can be done. This is because when parsing a single character argument, it is necessary to know if the corresponding argument requires an additional parameter.

According to the existing parsing logic, a command line like `eza -tmodified` is valid. If, as suggested by @WhyNotHugo, `eza` can't tell whether the `ra` after the letter `t` in `eza -ltra` is another two arguments or the parameters belonging to `t`. This is ambiguity.

**Comment by WhyNotHugo** [comment]
Quite true. I guess the only options are to chose one of the following:

- Remain compatible with exa.
- Remain compatible with ls.

**Comment by MartinFillon** [comment]
> I don't think this can be done. This is because when parsing a single character argument, it is necessary to know if the corresponding argument requires an additional parameter.

This is doable, and we have what needed but as Clap is incoming I dont wanna take time to modify it, you are free to do it yourself tho.

**Comment by teto** [comment]
I've also discovered recently `ls -p` which suffixes folders with a `/`, making them more distinct, and that eza lacks:
```
➜ eza -p                    
eza: Unknown argument -p
home on  boot-fixes via ❄️  IMPURE (dotfiles-shell-env) 
❌3 ❯ eza --version  
eza - A modern, maintained replacement for ls
v0.18.0 [+git]
https://github.com/eza-community/eza
```

**Comment by cafkafk** [comment]
> I've also discovered recently `ls -p` which suffixes folders with a `/`, making them more distinct, and that eza lacks:
> 
> ```
> ➜ eza -p                    
> eza: Unknown argument -p
> home on  boot-fixes via ❄️  IMPURE (dotfiles-shell-env) 
> ❌3 ❯ eza --version  
> eza - A modern, maintained replacement for ls
> v0.18.0 [+git]
> https://github.com/eza-community/eza
> ```

This should probably be a separate issue, seems useful, and easy to implement, although I'm not much for using a single letter flag for this

**Comment by MartinFillon** [comment]
> I've also discovered recently ls -p which suffixes folders with a /, making them more distinct, and that eza lacks:

`eza -F` works the same as it classifies which type of file it is, it then adds the `/` for directories and `*` for executables and '@' for links

**Comment by adrian-the-git** [comment]
>eza -F works the same

...it works the same unless you try something like `-Fl`.

The `-t` flag is my personal dealbreaking itch with this tool (I run `ls -lrt` hundreds of times per day). I don't want to develop muscle memory for `-lsnew` and find myself fighting with vanilla linux.

I haven't looked at the argument parsing code—is this kind of disambiguation for `ls` compatibility likely to be complicated to implement? I think we all have our repetitive behaviors with `ls`. If this kind of fix is unlikely to work with eza, I might go back to my ancient-but-functional color ls drop-in replacement.

**Comment by WhyNotHugo** [comment]
> I haven't looked at the argument parsing code—is this kind of disambiguation for ls compatibility likely to be complicated to implement?

I implemented it and it works on many cases, but there are edge cases where it can be ambiguous ([as mentioned in a comment above](https://github.com/eza-community/eza/issues/736#issuecomment-1922736980)).

**Comment by moonfruit** [comment]
> The `-t` flag is my personal dealbreaking itch with this tool (I run `ls -lrt` ~hundreds~ zillions of times per day). I don't want to develop muscle memory for `-lsnew` and find myself fighting with vanilla linux.

Yes. It itch me too. So I made a script to wrap eza: when `-lrt` is appear the script simply replace it with `-lsnew`, then send the argument to eza.

**Comment by genevieve-me** [comment]
Even if it's not `-t`, is there a possibility of adding some other short option for sorting by mtime? Like others have mentioned I used this *constantly* and I suppose I can add a new shell alias `lst` or `lat` but I would really love to see eza just stick closer to `ls` behavior here.

**Comment by angrox** [comment]
This is also a dealbreaker for me. I run `ls -lart` too often at too many systems where I just have ls instead of eza. 

Idea: compability env variable for those options? EZA_LS_COMPABILITY=1 ?

**Comment by cafkafk** [comment]
> This is also a dealbreaker for me. I run `ls -lart` too often at too many systems where I just have ls instead of eza.
> 
> Idea: compability env variable for those options? EZA_LS_COMPABILITY=1 ?

Our plan is to introduce a flag that makes eza ls compatible, this is gonna be done after the clap migration

**Comment by cafkafk** [comment]
A lot of specific feedback seems to get lost in this thread, if you have some feature request, make sure to create a separate issue for it, I just see "work as drop-in replacement" in the issue tracker, but there are suggestions here like a short flag for sorting by mtime that likely have a harder time being picked up by contributors from not having their own issue

**Comment by MartinFillon** [comment]
> Our plan is to introduce a flag that makes eza ls compatible, this is gonna be done after the clap migration

speaking of that if anyone wanna help review it as its a long standing pr we will accept the help

**Comment by mschwartz** [comment]
-lrt a gazillion times for me, too.  Muscle memory.

I am removing my alias ls=‘eza’

I will revisit when you fix this.

It’s not drop in replacement as advertized…

**Comment by moonfruit** [comment]
I wrote this simple wrapper script to deal with the hassle of `-lrt`. It’s not perfect, but it solves 90% of the problem and works really well. If you need it too, you can name it `eza` and put it in a directory that’s in your `PATH`. Just make sure this directory comes before the one containing the real `eza`, so the wrapper overrides the original. Alternatively, you can name it `ls` and place it before the real `ls` in your PATH.

```bash
#!/usr/bin/env bash
if [[ "$1" = "-lrt" ]]; then
    set -- "-lsnew" "${@:2}"
fi
exec /path/to/real/eza "$@"
```

**Comment by kesor** [comment]
> I wrote this simple wrapper script to deal with the hassle of `-lrt`.
@moonfruit

When you want something more comprehensive, try this https://gist.github.com/kesor/6e094187c85f05c30fc923d6923c1073

**Comment by moonfruit** [comment]
Thanks @kesor. I do have a more complicated script designed for myself. But I don’t think it’s suitable for most people. So I didn’t include it here, but shared a simpler script instead. I just wanted to solve this problem in a simple way for most people, without introducing too many other things.
